applications/GeoMechanicsApplication/tests/cpp_tests/test_equation_of_motion.cpp

```diff
@@ -183,4 +184,90 @@ KRATOS_TEST_CASE_IN_SUITE(CalculateStiffnessMatrixGivesCorrectResults, KratosGeo
     KRATOS_CHECK_MATRIX_NEAR(stiffness_matrix, expected_stiffness_matrix, 1e-4)
 }
 
+KRATOS_TEST_CASE_IN_SUITE(TheInternalForceVectorIsTheIntegralOfBTransposedTimesSigma, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const auto b_matrix                 = Matrix{ScalarMatrix{2, 8, 1.0}};
+    const auto b_matrices               = std::vector<Matrix>{b_matrix, b_matrix};
+    const auto stress_vector            = Vector{ScalarVector{2, 1.0}};
+    const auto stress_vectors           = std::vector<Vector>{stress_vector, stress_vector};
+    const auto integration_coefficients = std::vector<double>{0.25, 0.4};
+
+    const auto     expected_internal_force_vector = Vector{ScalarVector{8, 1.3}};
+    constexpr auto relative_tolerance             = 1.0e-6;
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(GeoEquationOfMotionUtilities::CalculateInternalForceVector(
+                                           b_matrices, stress_vectors, integration_coefficients),
+                                       expected_internal_force_vector, relative_tolerance)
+}
+
+// The following tests only raise errors when using debug builds
+#ifdef KRATOS_DEBUG
```

Comment:
Do we think these checks would ever be helpful for cases which are run in release? I like that this way it doesn't impact release performance, but I'm just putting the question here.

Reply:
Since the input for `GeoEquationOfMotionUtilities::CalculateInternalForceVector` is prepared by us (the programmers) and any inconsistencies in the input data are expected to affect previous operations, @WPK4FEM and myself decided that a debug check is adequate here. It's more like an assertion: this is never supposed to happen, but we do want to catch it at debug-time _if_ it ever happens. As you wrote yourself, the end-user (who uses a release build) doesn't pay the price of the additional checks, which benefits general performance.